src/components/ReactTags.js

```diff
@@ -387,7 +387,7 @@ class ReactTags extends Component {
     return tags.map((tag, index) => {
       return (
         <Tag
-          key={`${tag.id}-${index}`}
+          key={tag.key || tag.id}
```

Comment:
I don't believe we should have `tag.id` higher priority than `tag.key` for the following reasons:
1) most users this library are not sending a `key` property anyway (virtually no change for them)
2) react uses the `key` property to determine uniqueness already, so this is more in line with the "react way" of doing things